### PR TITLE
squeezelite: add new squeezelite package to sound/

### PR DESCRIPTION
**Pull request by squeezewrt**
Squeezelite (https://code.google.com/p/squeezelite/)
is a small headless squeezebox emulator for linux using alsa audio output.
It is aimed at supporting high quality audio including usb dac based output
at multiple sample rates including 44.1/48/88.2/96/176.4/192k/352.8/384kHz

There is a switchable option to compile the package with LIRC support. Another PR to add LIRC package will be made in the near future.

Signed-off-by: Andrew Kazakov <squeezewrt.ak@gmail.com>

### CODE REVIEW

**Comment by dangowrt** [comment]
Please squash the three commits into a single one can call it 'squeezelite: add new package'.
Also it doesn't make sense to include any dependency towards LIRC before the package is available. I I reckon you should rather update the squeezelite package once the LIRC package got merged as well.

**Comment by thess** [comment]
I have done some testing with this package and have some observations and concerns about how it is put together. Overall, it works pretty well once you figure out how to configure it. For starters, it forces the build of a number of codec libraries that are not included as dependencies due to its dynamic loading of the codecs. As a result if one installed the package via opkg, it will not work at all. You still need to install additional packages (without any information about which are needed) as well as tailor its config before things will work. There is a LuCI package proposed to manage the config but it too doesn't help you with the missing dependencies.

The Makefile will allow a custom build to have these libs pre-installed in the firmware image if you select them in via optional items but nothing is selected by default. The squeezelite package itself still has no package dependencies for these options in the control file. I think it is somewhat unusual to have image profile contents controlled by a specific package.

The dependence on LIRC is also a bit problematic. Both this package and the LIRC in oldpackages (0.8.7) are pre-configured for a specific IR interface and kernel config. LIRC itself is somewhat complicated to configure and deploy without kernel support and customized IR device profiles. Additionally, LIRC support is now included in recent kernels and the older version of LIRC only supports a limited set of devices. I think to integrate the latest LIRC releases (0.9.x) into OpenWrt needs additional kernel config options as well as specific configs for packages that want to use it.

Bottom line recommendations:
* A package with some built-in defaults and dependencies would be a better start.
* Remove LIRC support for now until it is better integrated (or more general).
* Add ALAC support to ffmpeg-mini (I've already done that #1349)

**Comment by squeezewrt** [comment]
OK

I remove all about LIRC from Makefile, startup script and default config file.
Squeezelite package now is always depends on the libffmpeg-mini and always compiled with -DFFMPEG switch
All codecs now enabled by default in the Makefile and initial config file. By the way, if the codec is not in the player, the decoding takes place in the server and player plays PCM

The corresponding changes in the luci-app-squeezelite package going to do in the near future

**Comment by thess** [comment]
A couple of things left to do

1. You commits need to be rebased/squashed into a single commit as Daniel mentioned in .-3. Don't forget your signed-off-by in the commit.
2. Please reference libffmpeg-audio-dec if you want ALAC. #1349 has be closed (not merged). It's your call whether this is the default or optional for size reasons. I think default 'no' would be optimal. Do not force the compilation of libffmpeg if not selected
3. Package dependencies are still incorrect. The package does not force the install of the any audio codec libs. You need to add them to DEPENDS conditional upon either package config or selection option.

**Comment by squeezewrt** [comment]
I'm totally confused. I'm not a professional, GIT for me - "Chinese puzzle". Why №1360? What do I do with it?
What do I do first? Combine commits to one commit? Add support for ALAC? Fix package dependencies? Recently I discovered a strange error in the dynamic loading of codec libraries. Maybe at first to fix it?
About package dependencies: I would not like to make the addition of all codecs to firmvare mandatory. All codecs + LuCI will not fit in the router with 4 MB of flash (my TL-TG3220, for example).

Porting skueezelite I started for fun. After I made the first pull reguest, fun was over, began the hassle. I do not know if I want to do it on :(

**Comment by thess** [comment]
My pull-request (#1360) is for the import of squeezelite with the dependencies conditional upon menuconfig options. If you don't want to own it, I'll be glad to take it on as I do want to see this package in OpenWrt. After it is imported, feel free to submit changes/updates to this repo via pull-requests.

If size is your main concern, the what codecs should be the default - MP3 & FLAC? Should OGG and AAC be optional like WMA/ALAC, etc. (btw - use ffmpeg-audio-codec to include ALAC). We'd need provide some further doc on how to install additional codecs and enable them.

What about DSD over PCM support - default on or off (or just built-in the base package)
What problem(s) did you see with dynamic lib loading?

The LuCI app should be handled separately in that repo after we have squeezelite working. Is it intended to be mandatory for squeezelite? Personally, I don't think so.

I am not so sure about how to handle LIRC at this time. Perhaps someone else could advise. It needs to be more general and not just specific to squeezelite and only one IR device. (Example: I would want to use my MCE remote for example which needs kernel support.)

Feel free to email me directly if you want to take this conversation off-line.

**Comment by squeezewrt** [comment]
Judging by the changes that you made in # 1360 (almost all codecs included, resampler), you are going to use more powerful than a conventional, router, hardware. My target is low-cost ($ 10-20) router and IR remote control necessarily.
As I wrote above, work on the project ceased to be fun and start - headache. If you so wish, you can take the project itself. Just delete my name from Makefile.
And I'll be back to work on my own unofficial feed with LIRC from oldpackages.

**Comment by thess** [comment]
Fair enough...

Just out of curiosity though - what codecs do you install on your $10-20 router? I'd like to have anyone installing the pre-built package (.ipk) to have something that works out-of-the-box. No ffmpeg, no soxr, subset of codecs like MP3 & FLAC as dependencies, others optional for custom builds.

**Comment by squeezewrt** [comment]
FLAC only. And this is a special assembly I do. If you install through .ipk it does not fit anything.

By the way, in the router can be not installed any codec (I wanted to do this by default). When a player connects to a server, it tells to server what codecs he had "on board." If, for example, need to play .ogg and codec is not installed in player, the server decodes the file "on the fly", and passes to the player WAV.
